driver-core/src/main/java/com/datastax/driver/core/Frame.java

```diff
@@ -201,19 +201,32 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf buffer, List<Object> ou
             // version first bit is the "direction" of the frame (request or response)
             version = version & 0x7F;
 
-            DecoderForStreamIdSize decoder = (version >= 3) ? decoderV3 : decoderV1;
+            DecoderForStreamIdSize decoder = decoderForStreamIdSize.get();
+            if (decoder == null) {
```

Comment:
> You're safe using a plain private field without any synchronization.

awesome, I thought that was the case but wasn't completely sure,  I'll nix the `AtomicReference`.